crates/bevy_transform/src/lib.rs

```diff
@@ -22,14 +27,27 @@ impl Plugin for TransformPlugin {
             .register_type::<Transform>()
             .register_type::<GlobalTransform>()
             // add transform systems to startup so the first update is "correct"
-            .add_startup_system_to_stage(startup_stage::POST_STARTUP, parent_update_system.system())
+            .add_startup_stage_after(
+                startup_stage::STARTUP,
+                stage::STARTUP_UPDATE_HIERARCHY,
+                SystemStage::parallel(),
+            )
+            .add_startup_system_to_stage(
+                stage::STARTUP_UPDATE_HIERARCHY,
+                parent_update_system.system(),
+            )
             .add_startup_system_to_stage(
                 startup_stage::POST_STARTUP,
                 transform_propagate_system::transform_propagate_system.system(),
             )
-            .add_system_to_stage(stage::POST_UPDATE, parent_update_system.system())
+            .add_stage_after(
+                bevy_app::stage::UPDATE,
+                stage::UPDATE_HIERARCHY,
+                SystemStage::parallel(),
+            )
```

Comment:
```suggestion
            .add_stage_before(
                bevy_app::stage::POST_UPDATE,
                stage::UPDATE_HIERARCHY,
                SystemStage::parallel(),
            )
```
I think this is a better placement for the stage.